### PR TITLE
Refactor inventory edit mode [SCI-4070]

### PR DESCRIPTION
**Pull request by okriuchykhin**
Jira ticket: [SCI-4070](https://biosistemika.atlassian.net/browse/SCI-4070)

### What was done
Refactor inventory edit mode

### CODE REVIEW

File: app/assets/javascripts/repositories/repository_datatable.js [hunk 1]
```diff
@@ -219,21 +214,77 @@ var RepositoryDatatable = (function(global) {
...
+  function checkAvailableColumns() {
+    $.ajax({
+      url: $(TABLE_ID).data('available-columns'),
+      type: 'GET',
+      dataType: 'json',
+      success: function(data) {
+        var columnsIds = data.columns;
+        var presentColumns = $(TABLE_ID).data('repository-columns-ids');
+        if (!_.isEqual(columnsIds.sort(), presentColumns.sort())) {
+          alert($(TABLE_ID).data('columns-changed'));
```
Comment (aignatov-bio) [inline app/assets/javascripts/repositories/repository_datatable.js]: I think you forgot to remove this alert

File: app/assets/javascripts/repositories/repository_datatable.js [hunk 2]
```diff
@@ -50,22 +46,11 @@ var RepositoryDatatable = (function(global) {
...
-  }
-
-  function changeToEditMode() {
-    currentMode = 'editMode';
-    // Table specific stuff
-    TABLE.button(0).enable(false);
-  }
-
   // Enable/disable edit button
   function updateButtons() {
```
Comment (aignatov-bio) [inline app/assets/javascripts/repositories/repository_datatable.js]: Maybe we can somehow refactor this function? Add general class for element which must be disabled.

File: app/assets/javascripts/repositories/repository_datatable.js (hunk 1 above)
Comment (okriuchykhin) [reply app/assets/javascripts/repositories/repository_datatable.js]: Why? It is a part of column checking logic.

File: app/assets/javascripts/repositories/repository_datatable.js (hunk 2 above)
Comment (okriuchykhin) [reply app/assets/javascripts/repositories/repository_datatable.js]: It won't work here due to complex switch on/off scenarios

**Comment by okriuchykhin** [comment]
Only one line at the time editing is supported

**Pull request merged**